pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -411,8 +487,13 @@ func isNoLongerOutdated(r *compv1alpha1.ComplianceRemediation) bool {
 
 func setRemediationStatus(r *compv1alpha1.ComplianceRemediation, errorApplying error, logger logr.Logger) {
 	if errorApplying != nil {
-		logger.Info("Remediation had an error")
-		r.Status.ApplicationState = compv1alpha1.RemediationError
+		if wasErrorOnOptionalRemediation(r, errorApplying) {
+			logger.Info("Optional remediation couldn't be applied")
+			r.Status.ApplicationState = compv1alpha1.RemediationNotApplied
```

Comment:
should we at least issue an event to let the admin know that the operation failed?

Reply:
yes, let's do that